### ISSUE TITLE
Implementation of cylinder primitive?

### ISSUE THREAD

**Issue by yudongqiu** [tunabrain/tungsten#44]
Hi Mr. Bitterli,

I was introduced to your renderer by a friend, and was impressed by the result and performance. As I'm trying to render some molecular pictures, a "cylinder" geometry would be very helpful for showing chemical bonds. Without any knowledge of computer graphics, I read the implementation of other primitives in `src/core/primitives` and tried to implement the cylinder primitive by mimicking them. My first draft was able to compile, but the cylinder is not showing up in the rendered results, except some reflections on a sphere. I have very little understanding of how the rendering works, so many of the functions may be done terribly wrong. If you have a little time, could you please take a look at it? (https://github.com/yudongqiu/tungsten/blob/master/src/core/primitives/Cylinder.cpp)

Any hint on my mistakes, or references that could help me understanding it, would be greatly appreciated. It would be also great that if you think writing it from scratch is better than looking at mine. 😄  Thanks a lot!

Best,

Yudong Qiu

**Comment by tunabrain** [comment]
Hi Yudong,

Thanks for your efforts! I decided to implement a cylinder primitive (both with and without end caps). I just pushed it to the master branch.

Here is an example scene:

![cylinders](https://user-images.githubusercontent.com/4923858/31835665-7aeac4dc-b5a0-11e7-83d9-ca7d48f7118f.png)

The scene file is attached here: [scene.zip](https://github.com/tunabrain/tungsten/files/1402931/scene.zip).

Let me know if this works for your purposes!

Thanks,
Benedikt
